frodo.js

```diff
@@ -5,88 +5,97 @@
  * for the extras/ plugins if they are loaded into a presentation.
```

Comment:
SCRIPT-I like how you've used promise chaining here, it makes the code flow very clear.